tests/test_native_serve.py

```
@@ -232,9 +359,92 @@ def test_request_id_passthrough(base_url: str, tr: TestResults):
+             f"got {r.status_code}: {data}")
+
+
+def test_malformed_wav_huge_frame_count(base_url: str, tr: TestResults):
+    """A WAV header claiming billions of frames fails fast with 400.
+
+    Crafted RF64: the ds64 sampleCount (0x100000000) is reported verbatim by
+    dr_wav over an 8-byte payload. The decoder derives the frame cap from the
+    payload, rejects the file, and the RIFF/RF64 magic keeps it out of the
+    raw-PCM16 fallback (issue #12: no gigabyte allocations, no garbage 200s).
+    """
+    def put(fmt, *vals):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `put`

(ANN202)